spec/routes/routes_spec.rb

```diff
@@ -14,4 +14,32 @@
       expect(get: "student-loans/non-existent-page").not_to be_routable
     end
   end
+
+  describe "service is unavailable" do
+    before do
+      allow(Rails.application.config.x).to receive(:maintenance_mode) { maintenance_mode }
+      Rails.application.reload_routes!
+    end
+
+    after do
+      allow(Rails.application.config.x).to receive(:maintenance_mode).and_call_original
```

Comment:
RSpec tears down the mocks at the end of each example, so does that make this unnecessary?

Reply:
Yeah, you're right. That was a hangover from when I was having trouble stubbing `ENV`. I'll bin it off

Reply:
No, wait, ignore that, I do have remove the stubs here, because the `after` block is still in the context of the tests, so I still need to remove the stub before reloading the routes, as the routes I've loaded in my `before` block will still remain in memory (as routes don't get reloaded at the start of every test)